source/SemanticModel.cpp

```diff
@@ -359,20 +366,81 @@ void SemanticModel::handleLoopGenerate(const LoopGenerateSyntax* syntax, SmallVe
     }
 }
 
-void SemanticModel::handleGenerateBlock(const MemberSyntax* syntax, SmallVector<const Symbol*>& results, const Scope* scope) {
+void SemanticModel::handleGenerateBlock(const MemberSyntax* syntax, SmallVector<const Symbol*>& results, const Scope* _scope) {
+    Scope *scope = alloc.emplace<Scope>(_scope);
     if (syntax->kind == SyntaxKind::GenerateBlock) {
         SmallVectorSized<const Symbol*, 8> children;
         for (const MemberSyntax* member : syntax->as<GenerateBlockSyntax>()->members) {
-            switch (member->kind) {
-                case SyntaxKind::HierarchyInstantiation:
-                    handleInstantiation(member->as<HierarchyInstantiationSyntax>(), children, scope);
-                    break;
-            }
+            handleGenerateItem(member, children, scope);
         }
-        results.append(alloc.emplace<GenerateBlock>(children.copy(alloc)));
+        results.append(alloc.emplace<GenerateBlock>(children.copy(alloc), scope));
     }
     else {
-        // TODO: handle simple member
+        handleGenerateItem(syntax, results, scope);
+    }
+}
+
+static ProceduralBlock::Kind proceduralBlockKindFromKeyword(Token kindKeyword) {
+    switch (kindKeyword.kind) {
+        case TokenKind::InitialKeyword: return ProceduralBlock::Initial;
+        case TokenKind::FinalKeyword:   return ProceduralBlock::Final;
+        case TokenKind::AlwaysKeyword:  return ProceduralBlock::Always;
+        case TokenKind::AlwaysCombKeyword:  return ProceduralBlock::AlwaysComb;
+        case TokenKind::AlwaysFFKeyword:  return ProceduralBlock::AlwaysFF;
+        case TokenKind::AlwaysLatchKeyword:  return ProceduralBlock::AlwaysLatch;
+        default:
+            ASSERT(false, "Unknown ProceduralBlock kind keyword: %s",
+                   kindKeyword.toString().c_str());
+            break;
+    }
+}
+
+void SemanticModel::handleProceduralBlock(const ProceduralBlockSyntax *syntax, SmallVector<const Symbol *>& results, const Scope* _scope) {
+    Scope *scope = alloc.emplace<Scope>(_scope);
+    SmallVectorSized<const Symbol*, 2> children;
+    //TODO handleStatement(syntax->statement, children, scope);
+    results.append(alloc.emplace<ProceduralBlock>(
+        children.copy(alloc), proceduralBlockKindFromKeyword(syntax->keyword), scope));
+}
+
+void SemanticModel::handleDataDeclaration(const DataDeclarationSyntax *syntax, SmallVector<const Symbol *>& results, Scope* scope) {
+    DataSymbol::DataSymbolModifiers modifiers;
+    modifiers.constM = std::find_if(syntax->modifiers.begin(), syntax->modifiers.end(), [](auto tk) {return tk.kind == TokenKind::ConstKeyword;}) != syntax->modifiers.end();
```

Comment:
This section is a little wasteful since we loop over every modifier for every modifier type. Not a huge deal since there won't ever be a lot of them, but to do the minimum work we'd just loop over syntax->modifiers once, switch() on the kind and then set the modifier member appropriately.